spec/request/legacy_urls_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe LegacyUrlsController, type: :request do
+  let(:resource) { create :work_version }
+
+  describe 'Scholarsphere v3 legacy URLs' do
+    before do
+      create :legacy_identifier,
+             version: 3,
+             old_id: 'old-v3-id-123',
+             resource: resource
+    end
+
+    it do
+      get '/concern/generic_works/old-v3-id-123'
+      expect(response).to redirect_to resource_path(resource.uuid)
+    end
+
+    context 'when given a nonexistent id' do
+      it do
+        expect {
+          get '/concern/generic_works/doesnt-exist'
+        }.to raise_error(ActiveRecord::RecordNotFound)
```

Comment:
@rschenk shouldn't this just be a 404?

Reply:
It will be, that's handled by a Rails middleware thing. All `ActiveRecord::RecordNotFound` are caught and turned into 404s automagically in productionesque environments